### PR TITLE
Add Expr, Stmt, Pat to schema.py

### PR DESCRIPTION
**Pull request by aibaars**
This PR adds classes for expressions, statements, and patterns to the schema. The code is most auto-generated by co-pilot. The definitions of the HIR datatypes can be found at https://github.com/rust-lang/rust-analyzer/blob/master/crates/hir-def/src/hir.rs#L169-L300

### CODE REVIEW

**Comment by paldepind** [comment]
Looks amazing! :tada: Having `AstNode` would also be very useful I think. I'm looking at setting up the boilerplate for the CFG and `AstNode` is used a bunch of places.

File: rust/schema.py [hunk 1]
```diff
@@ -31,30 +33,514 @@ class Location(Element):
...
+
+class Yield(Expr):
+    expr: optional[Expr]
+
+#     Yeet {
+#         expr: Option<ExprId>,
+#     },
+
+
+class Yeet(Expr):
```
Comment (redsun82) [inline rust/schema.py:272]: for the curious, this is an experimental feature that we're not probably going to really encounter https://doc.rust-lang.org/stable/unstable-book/language-features/yeet-expr.html

File: rust/schema.py [hunk 2]
```diff
@@ -31,30 +33,514 @@ class Location(Element):
...
+
+class Box(Expr):
+    expr: Expr
+#     UnaryOp {
+#         expr: ExprId,
+#         op: UnaryOp,
+#     },
+
+
+class UnaryExpr(Expr):
```
Comment (redsun82) [inline rust/schema.py:338]: I think you mean
```suggestion
class UnaryOp(Expr):
```

File: rust/schema.py [hunk 3]
```diff
@@ -31,30 +33,514 @@ class Location(Element):
...
+#         lhs: ExprId,
+#         rhs: ExprId,
+#         op: Option<BinaryOp>,
+#     },
+
+
+class BinaryOp(Expr):
+    lhs: Expr
+    rhs: Expr
+    op: optional[string]
```
Comment (redsun82) [inline rust/schema.py:353]: out of curiosity, do you know what does `op: None` mean?

**Review by redsun82 — COMMENTED** [review]
Some small naming nits and one probable typo, and some questions from my side, but this looks great as a starting point, thanks!

We should not forget when we do start extracting things to add some code snippets as examples to class docstrings. That helps us along the way, the QL writer then, and if we don't do it from the start we risk never really catching up

File: rust/schema.py (hunk 3 above)
Comment (aibaars) [reply rust/schema.py:353]: I have no idea, I am equally curious. Just haven't found the time to run an example expression through the pretty_print function.

File: rust/schema.py (hunk 1 above)
Comment (paldepind) [reply rust/schema.py:285]: Maybe we should just remove it for now? Given that it's highly experimental and placeholder syntax that is going to change in the future.

Comment (aibaars) [reply rust/schema.py:285]: Not sure. I could remove the class and translate `yeet` to `MissingExpr` in the extractor.

**Pull request merged**